client/state/sites/middleware.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * Internal dependencies
+ */
+import sitesList from 'lib/sites-list';
+import { receiveSites } from 'state/sites/actions';
+
+export default ( { dispatch } ) => {
+	const sites = sitesList();
+	sites.on( 'change', () => {
+		dispatch( receiveSites( sites.get() ) );
```

Comment:
>Could we have issues with this `sites.get()` not being pure?

This should be handled the same way as how we [receive sites into Redux from `sites-list` in the `my-sites` controller](https://github.com/Automattic/wp-calypso/blob/6bb2e62f614bd6d58828b27e8db87a36d4482062/client/my-sites/controller.js#L169-L171). Which is: Received sites will have properties [picked using "valid" site keys from the state schema](https://github.com/Automattic/wp-calypso/blob/6bb2e62f614bd6d58828b27e8db87a36d4482062/client/state/sites/reducer.js#L66), and computed properties merged into when selecting [using the `getSite` selector](https://github.com/Automattic/wp-calypso/blob/6bb2e62f614bd6d58828b27e8db87a36d4482062/client/state/sites/selectors.js#L42-L70).